### PR TITLE
Reset origText in verifySignature before the retry

### PR DESCRIPTION
**Pull request by DmitriyMV**
This is partial fix for #231

### CODE REVIEW

File: crypto/signature.go [hunk 1]
```diff
@@ -261,12 +261,22 @@ func verifySignature(
 				return time.Unix(verifyTime, 0)
 			}
 
+			seeker, ok := origText.(io.ReadSeeker)
+			if !ok {
+				return nil, newSignatureFailed(errors.New("message reader do not support seeking, cannot retry signature verification"))
```
Comment (marinthiercelin) [inline crypto/signature.go:266]:
```suggestion
				return nil, newSignatureFailed(errors.Wrap(err, "gopenpgp: message reader do not support seeking, cannot retry signature verification"))
```
We should prefix the error message with the module name. And I would include the original error.

File: crypto/signature.go [hunk 2]
```diff
@@ -261,12 +261,22 @@ func verifySignature(
...
+			}
+
+			_, err = seeker.Seek(0, io.SeekStart)
+			if err != nil {
+				return nil, newSignatureFailed(err)
```
Comment (marinthiercelin) [inline crypto/signature.go:271]: We should wrap errors with the context.
```suggestion
				err = errors.Wrap(err, "gopenpgp: could not rewind the data reader.")
				return nil, newSignatureFailed(err)
```
Can you also add this to the error from `signatureReader.Seek(0, io.SeekStart)` below ? It seems we forgot to wrap it.

Comment (DmitriyMV) [reply crypto/signature.go]: Thanks fixed!

File: crypto/signature.go (hunk 1 above)
Comment (DmitriyMV) [reply crypto/signature.go]: Fixed!

**Review by marinthiercelin — APPROVED** [review]

**Pull request merged**